webapp/src/main/java/rocks/metaldetector/security/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,64 @@
+package rocks.metaldetector.security;
+
+import io.jsonwebtoken.Claims;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.authentication.AuthenticationDetailsSource;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import rocks.metaldetector.persistence.domain.user.AbstractUserEntity;
+import rocks.metaldetector.persistence.domain.user.UserRepository;
+import rocks.metaldetector.support.JwtsSupport;
+import rocks.metaldetector.support.exceptions.ResourceNotFoundException;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+import static rocks.metaldetector.service.user.UserErrorMessages.USER_WITH_ID_NOT_FOUND;
+
+@Slf4j
+@Component
+@AllArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+  private final JwtsSupport jwtsSupport;
+  private final UserRepository userRepository;
+  private final AuthenticationDetailsSource<HttpServletRequest, ?> authenticationDetailsSource;
+
+  @Override
+  protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+    try {
+      String token = parseHeader(request);
+      if (token != null) {
+        Claims claims = jwtsSupport.getClaims(token);
+        AbstractUserEntity user = userRepository.findByPublicId(claims.getSubject())
+            .orElseThrow(() -> new ResourceNotFoundException(USER_WITH_ID_NOT_FOUND.toDisplayString()));
+
+        UsernamePasswordAuthenticationToken authentication = new UsernamePasswordAuthenticationToken(user, null, user.getAuthorities());
+        authentication.setDetails(authenticationDetailsSource.buildDetails(request));
```

Comment:
why not passing the `WebAuthenticationDetailsSource` directly here and removing the bean?

```suggestion
        authentication.setDetails(new WebAuthenticationDetailsSource().buildDetails(request));
```

Reply:
Hm why? Can't test it as nice then and a new instance that has to be garbage collected at some point is created on every server interaction.